lib/cn/csched_sp3_work.c

```diff
@@ -265,9 +292,11 @@ sp3_work_wtype_idle(
     kvsets = cn_ns_kvsets(ns);
 
     if (cn_node_isroot(tn)) {
-        *action = CN_ACTION_SPILL;
-        *rule = CN_RULE_RSPILL;
-        return kvsets;
```

Comment:
This was an atrocious bug that could really corrupt the tree, but was hard to hit unless you had tombs/ptombs in the root node and some weird ingest patterns.
It was really difficult to debug, which is why I replaced the workid from dgen to compaction work struct - when the bug hit it was really confusing because in sp3_work() the workid which should have been zero had the correct dgen in it!